docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,339 @@
+
+### New Messages
+In what follows, we describe the new messages (i.e., `MsgRegisterConsumerChain`, `MsgInitializeConsumerChain`, and `MsgUpdateConsumerChain`)
+that Permissionless ICS introduces, and on how those can be used.
+Then, we describe how to utilize these messages with our existing codebase.
+
+#### Register a Consumer Chain
+We first have to register an Opt In chain before launching it. This is done through the following message.
+```protobuf
+message MsgRegisterConsumerChain {
+  // the title of this launch 
+  string title;
+  // the description of this consumer chain
+  string description;
+  // the chain id of the new consumer chain
+  string chain_id;
+  // the owner of this consumer chain
+  string owner_address;
+}
+```
+
+This response of this message contains a single `string`, that is the `consumerId` for this registered consumer chain and initializes
+a consumer chain in its registered phase. With the returned `consumerId`, validators can already opt in on the consumer
+chain to show their potential interest on the chain. Additionally, a front-end ICS launchpad can also present
+this chain. Additionally, this allows consumer chains to show that they are interested in joining ICS even though,
+they might not yet know the specific ICS parameters they would like to use (see `MsgInitializeConsumerChain`). 
+
+This message contains the `owner_address` that corresponds to the address that would be able to initialize or later update this consumer chain.
+We store the owner address of each Opt In consumer chain by creating an association between `consumerId`s and `owner_address`es.
+Top N chains do not have an `onwer_address` because they can only be modified through governance proposals. 
+
+To prevent an attacker spamming the system by creating bogus consumer chains, we set a fixed cost for sending a `MsgRegisterConsumerChain` (configurable via a parameter).
+
+#### Launch a Consumer Chain
+To move a consumer chain to its initialized phase, we issue a `MsgInitializeConsumerChain` message that is as follows:
+
+```protobuf
+message MsgInitializeConsumerChain {
+  // tx signer address
+  string signer;
+  // consumer id of the to-be-updated consumer chain
+  string consumer_id;
+  // the initialization record that contains more specific parameters for the upcoming chain
+  InitializationRecord record;
+}
+```
+
+where `InitializationRecord` contains the following:
+```protobuf
+message InitializationRecord {
+  // ---------- ---------- ----------
+  // Following fields used to track metadata about the to-be-launched consumer chain before it launches.
+  // ---------- ---------- ----------
+  
+  // the title of the chain to-be-launched 
+  string title;
+  // the description of the chain to-be-launched
+  string description;
+  // the chain id of the new consumer chain
+  string chain_id;
+  // consumer id of the to-be-launched chain
+  string consumer_id;
```

Comment:
Why are these fields part of the initialization record? More important, why are these fields part of the MsgInitializeConsumerChain?

Reply:
After the offline discussion we had the other day, this has changed substantially so I'm resolving this as outdated.